src/components/Button/BistateButton.tsx

```diff
@@ -0,0 +1,76 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import Button, { ButtonProps, FabricButtonType } from './Button';
+
+export type CustomizableProps = 'color' | 'text' | 'ariaLabel' | 'icon' | 'iconPosition' | 'className';
+
+export type CustomizableButtonProps = Pick<ButtonProps, CustomizableProps>;
+
+export interface BistateButtonOwnProps {
+  activeProps: CustomizableButtonProps;
+}
+
+export type BistateButtonProps = BistateButtonOwnProps & ButtonProps;
```

Comment:
I lost track of the type manipulation in lines 6-14. Can you explain a bit?

Reply:
So BistateButtonProps is just ButtonProps + 1 prop `activeProps` which is a subset of ButtonProps; the 'customizable button props' that you might want to vary when the button is 'active'

Reply:
Can you think of better naming / an easier interface. It's obvious to me because I wrote it but I can see how it could be confusing

Reply:
Would lines 10-14 be equivalent to this?

```
export interface BistateButtonProps extends BistateButtonProps {
  activeProps: CustomizableButtonProps;
}
```

Reply:
Yes but we cant extend ButtonProps because its a type sadly

`An interface may only extend a class or another interface.`

Reply:
Aaaah.